src/index.js

```diff
@@ -73,35 +70,35 @@ function makeOptional(val) {
   }
 }
 
-function SpanCreator({ tracer, serviceName, kind }) {
-  return class Span {
-    _constructedFromOutside(options) {
-      return (
-        typeof options.traceId === "object" &&
-        typeof options.traceId.spanId === "string"
-      );
-    }
+function SpanCreator({ otTracer, tracer, serviceName, kind, sampler }) {
+  return class Span extends OpenTracing.Span {
     _getTraceId(options) {
-      // construct from give traceId
-      if (this._constructedFromOutside(options)) {
-        const { traceId, parentId, spanId, sampled } = options.traceId;
-        return new TraceId({
-          traceId: makeOptional(traceId),
-          parentId: makeOptional(parentId),
-          spanId: spanId,
-          sampled: makeOptional(sampled)
-        });
-      }
-
       // construct with parent
-      if (options.childOf !== null && typeof options.childOf === "object") {
-        const parent = options.childOf;
+      // for "references" see: https://github.com/opentracing/opentracing-javascript/blob/d2c1c6a5a50439bd769d02c7f4638293db40140a/src/tracer.ts#L79-L93
+      const childOf =
+        options.references &&
+        options.references.find(
+          r => r.type() === OpenTracing.REFERENCE_CHILD_OF
+        );
+      const followsFrom =
+        options.references &&
+        options.references.find(
+          r => r.type() === OpenTracing.REFERENCE_FOLLOWS_FROM
+        );
+      const parent = childOf || followsFrom; // prefer "child-of" refs over "follows-from" refs as "parent"
+      const parentCtx = parent && parent.referencedContext();
+      if (parentCtx) {
+        // with newer zipkin version switch to:
```

Comment:
Can you explain this comment here?

Reply:
the passed-in `references` array (which gets partly constructed or enhanced here: https://github.com/opentracing/opentracing-javascript/blob/d2c1c6a5a50439bd769d02c7f4638293db40140a/src/tracer.ts#L81-L92) can contain references to `child-of` relationships and to `follows-from` relationships.
IMHO, `child-of` is a better fit for a `parent` here and should be used if present; if it is not present a `follows-from` relationship should be used as `parent`...

Reply:
I agree with your conclusion, but what does this have to do with a newer zipkin version?